### PR TITLE
Bug 61528 - Pivot Table enhancements, new example

### PR DESCRIPTION
**Pull request by rk-poi**
here is the pull request for Bug 61528

### CODE REVIEW

**Comment by pjfanning** [comment]
Thanks - could you add some junit tests?

File: src/ooxml/testcases/org/apache/poi/xssf/usermodel/BaseTestXSSFPivotTable.java [hunk 1]
```diff
@@ -296,4 +313,45 @@ public void testPivotTableSheetNamesAreCaseInsensitive() {
...
+     * Verify that it's not possible to create a col label outside of the referenced area.
+     */
+    @Test
+    public void testAddColLabelOutOfRangeThrowsException() {
+        int columnIndex = 5;
+
+        try {
+            pivotTable.addColLabel(columnIndex);
+        } catch(IndexOutOfBoundsException e) {
+            return;
```
Comment (pjfanning) [inline src/ooxml/testcases/org/apache/poi/xssf/usermodel/BaseTestXSSFPivotTable.java]: Using `return` in the middle of a method is against most coding standards
Could you move to fail to the try block? Eg fail("expected an exception to be thrown")

File: src/examples/src/org/apache/poi/xssf/usermodel/examples/CreatePivotTable2.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        AreaReference source = new AreaReference("A1:E7", SpreadsheetVersion.EXCEL2007);
+        CellReference position = new CellReference("H1");
+        // Create a pivot table on this sheet, with H5 as the top-left cell..
+        // The pivot table's data source is on the same sheet in A1:D4
+        XSSFPivotTable pivotTable = sheet.createPivotTable(source, position);
+        //Configure the pivot table
+        //Use first column as row label
+        pivotTable.addRowLabel(0);
+        //Sum up the second column
+//        pivotTable.addColumnLabel(DataConsolidateFunction.SUM, 1);
```
Comment (pjfanning) [inline src/examples/src/org/apache/poi/xssf/usermodel/examples/CreatePivotTable2.java]: can we remove this line?

File: src/examples/src/org/apache/poi/xssf/usermodel/examples/CreatePivotTable2.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+import org.apache.poi.ss.usermodel.DataConsolidateFunction;
+import org.apache.poi.ss.usermodel.DataFormat;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.util.AreaReference;
+import org.apache.poi.ss.util.CellReference;
+import org.apache.poi.xssf.usermodel.XSSFPivotTable;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+public class CreatePivotTable2 {
```
Comment (pjfanning) [inline src/examples/src/org/apache/poi/xssf/usermodel/examples/CreatePivotTable2.java:38]: Can you add some javadoc about what the sample does?

File: src/ooxml/testcases/org/apache/poi/xssf/usermodel/BaseTestXSSFPivotTable.java (hunk 1 above)
Comment (rk-poi) [reply src/ooxml/testcases/org/apache/poi/xssf/usermodel/BaseTestXSSFPivotTable.java]: I have copied the function testAddRowLabelOutOfRangeThrowsException at this class and have changed the content ... :(
I have to looking for an example to change this function.

**Pull request closed without merging**